http4s/client/src/main/scala/endpoints4s/http4s/client/Endpoints.scala

```diff
@@ -243,23 +243,40 @@ trait EndpointsWithCustomErrors
       responseA(sc, hs)
         .map(_.andThen(_.map(_.asLeft[B])))
         .orElse(responseB(sc, hs).map(_.andThen(_.map(_.asRight[A]))))
-
-  //#endpoint-type
-  type Endpoint[A, B] = Kleisli[Effect, A, B]
+   //#endpoint-type
+  trait Endpoint[A, B] {
+    /** This method returns a Resource[Effect,B] unlike sendAndConsume this is always safe to use in regard to lazyness
+      *
+      * @param request
+      * @return
+      */
+
+    def send(request: A): Resource[Effect, B]
+
+    /** Note that this method CAN be unsafe if a lazy resource is returned as the Response due to closing the underlying Resource
+      *
+      * @param request
+      * @return
+      */
+    def sendAndConsume(request: A): Effect[B] = send(request).use(effect.pure)
+
+    @Deprecated(since = "5.0.0: This calls sendAndConsume which is potentially unsafe. Use sendAndConsume if you don't suffer any issues in regard to laziness, else use send")
```

Comment:
```suggestion
    @deprecated(since = "6.0.0", message = "This calls sendAndConsume which is potentially unsafe. Use sendAndConsume if you don't suffer any issues in regard to laziness, else use send")
```